pkg/resources/management.cattle.io/v3/fleetworkspace/validator_test.go

```diff
@@ -0,0 +1,100 @@
+package fleetworkspace
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/webhook/pkg/admission"
+	"github.com/rancher/wrangler/pkg/generic/fake"
+	admissionv1 "k8s.io/api/admission/v1"
+	v1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+func TestAdmit(t *testing.T) {
+	nsName := "test"
+
+	nsExistMutator := func() Mutator {
+		ctrl := gomock.NewController(t)
+		mockNamespaceController := fake.NewMockNonNamespacedControllerInterface[*v1.Namespace, *v1.NamespaceList](ctrl)
+		mockNamespaceController.EXPECT().Create(gomock.Any()).Return(nil, errors.NewAlreadyExists(schema.GroupResource{}, nsName))
+		mockNamespaceController.EXPECT().Get(gomock.Any(), gomock.Any()).Return(&v1.Namespace{}, nil)
+
+		return Mutator{
+			namespaces: mockNamespaceController,
+		}
+	}
+
+	newNsMutator := func() Mutator {
+		ctrl := gomock.NewController(t)
+
+		mockNamespaceController := fake.NewMockNonNamespacedControllerInterface[*v1.Namespace, *v1.NamespaceList](ctrl)
+		mockNamespaceController.EXPECT().Create(gomock.Any()).Return(&v1.Namespace{}, nil)
+		mockNamespaceController.EXPECT().Get(gomock.Any(), gomock.Any()).Times(0)
+		mockRoleBindingController := fake.NewMockControllerInterface[*rbacv1.RoleBinding, *rbacv1.RoleBindingList](ctrl)
+		mockRoleBindingController.EXPECT().Create(gomock.Any())
+		mockClusterRoleBindingController := fake.NewMockNonNamespacedControllerInterface[*rbacv1.ClusterRoleBinding, *rbacv1.ClusterRoleBindingList](ctrl)
+		mockClusterRoleBindingController.EXPECT().Create(gomock.Any())
+		mockClusterRoleController := fake.NewMockNonNamespacedControllerInterface[*rbacv1.ClusterRole, *rbacv1.ClusterRoleList](ctrl)
+		mockClusterRoleController.EXPECT().Create(gomock.Any())
+
+		return Mutator{
+			namespaces:          mockNamespaceController,
+			rolebindings:        mockRoleBindingController,
+			clusterrolebindings: mockClusterRoleBindingController,
+			clusterroles:        mockClusterRoleController,
+		}
+	}
```

Comment:
These functions don't need to be local variables in the test function. They can be moved out into regular functions that take in `t *testing.T` as a parameter or `ctrl *gomock.Controler`

Reply:
done in https://github.com/rancher/webhook/pull/260/commits/bc3715088473c65ec63c87a6da489240e56dbaef